plugin.js

```diff
@@ -39,7 +39,7 @@ function plugin (fastify, options, pluginRegistrationDone) {
     )
   }
 
-  fastify.decorateRequest('session', getSession())
+  fastify.decorateRequest('session', { getter: () => getSession() })
```

Comment:
I'd rather avoid arrow functions where possible:

```suggestion
  fastify.decorateRequest('session', { getter () { return getSession() } })
```